src/team_lib/mhba/xccl_mhba_team.c

```diff
@@ -458,6 +473,14 @@ xccl_status_t xccl_mhba_team_create_post(xccl_tl_context_t  *context,
 fail_after_transpose_reg:
     ibv_dereg_mr(mhba_team->transpose_buf_mr);
     free(mhba_team->transpose_buf);
+fail_ptr_malloc:
+    for (i = 0; i < MAX_OUTSTANDING_OPS; i++) {
+        xccl_mhba_op_t *op = &mhba_team->node.ops[i];
+        free(op->recv_umr_data);
```

Comment:
we can have mem corruption here in the cleanup flow: if failure happens on i=0 above then here you will call free on some undefined pointers for i != 0.